pkg/build/builder/common.go

```diff
@@ -82,3 +89,107 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds of the same originalTag.
+func randomBuildTag(originalTag string) string {
+	repo, _ := parsers.ParseRepositoryTag(originalTag)
+	suffix := fmt.Sprintf("%08x", rand.Uint32())
```

Comment:
I started using [`namer.GetName`](https://github.com/openshift/origin/blob/86a15eb95324991b0de57de04e71b29cec5ab63f/pkg/util/namer/namer.go#L15), if you will remember. But it doesn't do what we need here: `namer.GetName` adds a `-` between the base name and the suffix, and it will mangle the output with a hash if it is too long.
It also doesn't handle generating a random suffix.

The intended behavior here is to use the original repo, perhaps truncated if exceedingly long, plus a random 8-byte tag.
The idea of keeping the repo name is to facilitate recognizing the image that is being built, while the random tag allow us to uniquely and unambiguously identify the image that we need to tag with the user-provided repo:tag before push.

Please refer to the test cases in https://github.com/openshift/origin/pull/6715/files#diff-3fa9b8fb92546769141459a5a60a0193R56


@liggitt what "generate name util" are you referring to?